libmscore/read206.cpp

```diff
@@ -1795,8 +1795,14 @@ static void readLyrics(Lyrics* lyrics, XmlReader& e)
             delete _verseNumber;
             }
       lyrics->setAutoplace(true);
-      lyrics->setOffset(QPointF());
-//TODO-offset      lyrics->setOffset(QPointF());
+      if (!lyrics->isStyled(Pid::OFFSET) && !e.pasteMode()) {
+            // fix offset for pre-3.1 scores
+            // 2.x and earlier: y offset was relative to staff; x offset was relative to center of notehead
+            lyrics->rxoffset() -= lyrics->symWidth(SymId::noteheadBlack) * 0.5;
+            //lyrics->ryoffset() -= lyrics->placeBelow() && lyrics->staff() ? lyrics->staff()->height() : 0.0;
+            lyrics->setPlacement(Placement::ABOVE);
```

Comment:
Is really every y offset to mean placement above? IMHO only any (larger) negative y-offset, others (small and/or positive) most probablyhad been added to avoid collisions (and so are no longer needed, using autoplace)

Reply:
Not to worry, this is only temporarily setting offset to above, so adjustPlacement() knows that the original offset is relative to the top of staff (because that is how the offset is actually stored in the file - large positive offsets).  adjustPlacement will then move the element below.

I suppose there is some risk here, that for whatever reason adjustPlacement won't be able to figure out the lyric is really below, and may leave the placement set to above.  So I could be more conservative and only do any of this if the offset is in fact negative.  I'll look into that.

Reply:
So it turns out there is no way to avoid doing this, not if we want to honor the offset.  That's because, for better or for worse, 2.x wrote all offsets relative to the top staff line - which is to say, lyrics were treated as if they had above placement (because we didn't yet have the concept of below placement).  So if we want to honor any offset at all, we have to trust adjustPlacement() to convert it from above to below.

Again, in 3.0, we always throw away all offsets on lyrics, so whether you were moving them above the staff or just fine tuning the position below the staff a little, that info was being ignored even if you declined the reset.  So this change is going to be extremely beneficial.  Worst case there is there turn out to be some lyrics where for whatever reason we get the adjustment wrong and then you have to reset manually, or just decide to re-import and take the reset.